flatgfa/src/packedseq.rs

```diff
@@ -312,8 +419,62 @@ fn test_getter_setter() {
         Nucleotide::C,
         Nucleotide::C,
     ]);
-    assert_eq!(vec.get(0), Nucleotide::A);
-    assert_eq!(vec.get(1), Nucleotide::A);
+    assert_eq!(vec.as_ref().get(0), Nucleotide::A);
+    assert_eq!(vec.as_ref().get(1), Nucleotide::A);
     vec.set(1, Nucleotide::G);
-    assert_eq!(vec.get(1), Nucleotide::G);
+    assert_eq!(vec.as_ref().get(1), Nucleotide::G);
+}
+
+#[test]
+fn test_export_import_simple() {
+    let vec = PackedSeqStore::create(vec![
+        Nucleotide::A,
+        Nucleotide::C,
+        Nucleotide::T,
+        Nucleotide::G,
+    ]);
+    let input = vec.as_ref();
+    let filename = "capra_test_file";
+    let num_bytes = total_bytes(&input);
+    let mut mem = map_new_file(filename, num_bytes as u64);
+    let mut buf = vec![0u8; num_bytes];
+    let buf_ref = &mut buf;
+    input.dump(buf_ref);
+    mem[..buf_ref.len()].copy_from_slice(buf_ref);
+    let result: &[u8] = &mem;
+    let output = PackedSeqView::view(result);
+    assert_eq!(input.get(0), output.get(0));
+    assert_eq!(input.get(1), output.get(1));
+    assert_eq!(input.get(2), output.get(2));
+    assert_eq!(input.get(3), output.get(3));
+    std::mem::drop(mem);
+    let _ = std::fs::remove_file(filename);
+}
```

Comment:
Having through about this a little more, maybe the sensible thing to do here would be to test round-tripping through a `Vec<u8>` but never actually write that stuff to a file. That's almost as good but excludes the actual memory-mapping step, which is unlikely to be a problem. Basically, the idea would be that this "internal" test would test everything _except_ opening and closing files, basically, but everything else.

This would at least avoid the potential annoyance of creating and deleting temporary files.

Would this make sense? It would at least keep this test code a teensy bit simpler, and leave the actual filename handling to the CLI tests if we add those.